include/fixed_containers/fixed_deque.hpp

```diff
@@ -59,35 +65,41 @@ struct AbortChecking
 };
 }  // namespace fixed_containers::fixed_deque_customize
 
-namespace fixed_containers
+namespace fixed_containers::fixed_deque_detail
 {
+// FixedDeque<T> should carry the properties of T. For example, if T fulfils
+// std::is_trivially_copy_assignable<T>, then so should FixedDeque<T>.
+// This is done with concepts. However, at the time of writing there is a compiler bug
+// that is preventing usage of concepts for destructors: https://bugs.llvm.org/show_bug.cgi?id=46269
+// [WORKAROUND-1] due to destructors: manually do the split with template specialization.
+// FixedDequeBase is only used for avoiding too much duplication for the destructor split
 template <typename T,
           std::size_t MAXIMUM_SIZE,
-          fixed_deque_customize::FixedDequeChecking CheckingType =
-              fixed_deque_customize::AbortChecking<T, MAXIMUM_SIZE>>
-class FixedDeque
+          fixed_deque_customize::FixedDequeChecking CheckingType>
```

Comment:
If someone were to use the deque to implement a ring buffer - is that intended to be an adapter on top of the deque?

Reply:
This is a customization point for error handling. By default it just `std::abort()`s, but users can provide their own `Checking` that has exception and/or aborts etc.
It is not an adapter like https://en.cppreference.com/w/cpp/container/deque and https://en.cppreference.com/w/cpp/container/stack .

Note that currently the above two are not `constexpr` so there is a dedicated `FixedStack`(and `FixedDeque` will allow a similar `FixedQueue`) https://github.com/teslamotors/fixed-containers/blob/b3b17722fa79320bd17018cd61c6637296634c53/test/fixed_stack_test.cpp#L12